Looks like the uw-imap doesn't build on `darwin`:

```
If you get No such file error messages for files x509.h, ssl.h,
pem.h, buffer.h, bio.h, and crypto.h, that means that OpenSSL
is not installed on your system. Either install OpenSSL first
or build with command: make osx SSLTYPE=none
`cat CCTYPE` -c `cat CFLAGS` `cat OSCFLAGS` -c osdep.c
In file included from osdep.c:49:
In file included from ./env_unix.c:260:
In file included from ./auths.c:1:
./kerb_mit.c:30:10: fatal error: 'gssapi/gssapi_generic.h' file not found
#include <gssapi/gssapi_generic.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[3]: *** [Makefile:926: osdep.o] Error 1
make[3]: Leaving directory '/private/tmp/nix-build-uw-imap-2007f.drv-0/imap-2007f/c-client'
make[2]: *** [Makefile:591: osx] Error 2
make[2]: Leaving directory '/private/tmp/nix-build-uw-imap-2007f.drv-0/imap-2007f/c-client'
make[1]: *** [Makefile:688: OSTYPE] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-uw-imap-2007f.drv-0/imap-2007f'
make: *** [Makefile:427: osx] Error 2
error: builder for '/nix/store/wkif89sa6zbz001pv78i0zph687dd5iv-uw-imap-2007f.drv' failed with exit code 2;
       last 10 log lines:
       > #include <gssapi/gssapi_generic.h>
       >          ^~~~~~~~~~~~~~~~~~~~~~~~~
       > 1 error generated.
       > make[3]: *** [Makefile:926: osdep.o] Error 1
       > make[3]: Leaving directory '/private/tmp/nix-build-uw-imap-2007f.drv-0/imap-2007f/c-client'
       > make[2]: *** [Makefile:591: osx] Error 2
       > make[2]: Leaving directory '/private/tmp/nix-build-uw-imap-2007f.drv-0/imap-2007f/c-client'
       > make[1]: *** [Makefile:688: OSTYPE] Error 2
       > make[1]: Leaving directory '/private/tmp/nix-build-uw-imap-2007f.drv-0/imap-2007f'
       > make: *** [Makefile:427: osx] Error 2
       For full logs, run 'nix log /nix/store/wkif89sa6zbz001pv78i0zph687dd5iv-uw-imap-2007f.drv'.
error: 1 dependencies of derivation '/nix/store/lkp364krmw1wksmb720lxs46xhawvrgx-php-imap-8.1.11.drv' failed to build
```

--- a/pkgs/development/interpreters/php/8.0.nix
+++ b/pkgs/development/interpreters/php/8.0.nix
@@ -21,6 +21,7 @@ base.withExtensions ({ all, ... }: with all; ([
   gettext
   gmp
   iconv
+  imap
   intl
   ldap
   mbstring
@@ -49,4 +50,4 @@ base.withExtensions ({ all, ... }: with all; ([
   xmlwriter
   zip
   zlib
-] ++ lib.optionals (!stdenv.isDarwin) [ imap ]))
+]))

--- a/pkgs/development/interpreters/php/8.1.nix
+++ b/pkgs/development/interpreters/php/8.1.nix
@@ -21,6 +21,7 @@ base.withExtensions ({ all, ... }: with all; ([
   gettext
   gmp
   iconv
+  imap
   intl
   ldap
   mbstring
@@ -49,4 +50,4 @@ base.withExtensions ({ all, ... }: with all; ([
   xmlwriter
   zip
   zlib
-] ++ lib.optionals (!stdenv.isDarwin) [ imap ]))
+]))

--- a/pkgs/top-level/php-packages.nix
+++ b/pkgs/top-level/php-packages.nix
@@ -22,6 +22,7 @@
 , libffi
 , libiconv
 , libjpeg
+, libkrb5
 , libpng
 , libsodium
 , libwebp
@@ -342,10 +343,8 @@ lib.makeScope pkgs.newScope (self: with self; {
         }
         {
           name = "imap";
-          buildInputs = [ uwimap openssl pam pcre2 ];
-          configureFlags = [ "--with-imap=${uwimap}" "--with-imap-ssl" ];
-          # uwimap doesn't build on darwin.
-          enable = (!stdenv.isDarwin);
+          buildInputs = [ uwimap openssl pam pcre2 libkrb5 ];
+          configureFlags = [ "--with-imap=${uwimap}" "--with-imap-ssl" "--with-kerberos" ];
         }
         {
           name = "intl";

